docs/standard/serialization/system-text-json-converters-how-to.md

```diff
@@ -0,0 +1,745 @@
+}
+```
+
+The code to serialize and deserialize `WeatherForecastWithConverter` doesn't require the use of `JsonSerializeOptions.Converters`:
+
+```csharp
+string json = JsonSerializer.Serialize(weatherForecastWithConverter);
+```
+
+```csharp
+weatherForecast = JsonSerializer.Deserialize<WeatherForecastWithConverter>(json);
+```
+
+### Sample registration - [JsonConverter] on a type
+
+Here's code that creates a struct and applies the `[JsonConverter]` attribute to it:
+
+```csharp
+[JsonConverter(typeof(TemperatureConverter))]
+public struct Temperature
+{
+    public Temperature(int degrees, bool celsius)
+    {
+        _degrees = degrees;
+        _isCelsius = celsius;
+    }
+    private bool _isCelsius;
+    private int _degrees;
+    public int Degrees => _degrees;
+    public bool IsCelsius => _isCelsius; 
+    public bool IsFahrenheit => !_isCelsius;
+    public override string ToString() =>
+        $"{_degrees.ToString()}{(_isCelsius ? "C" : "F")}";
+    public static Temperature Parse(string input)
+    {
+        int degrees = int.Parse(input.Substring(0, input.Length - 1));
+        bool celsius = (input.Substring(input.Length - 1) == "C");
+        return new Temperature(degrees, celsius);
+    }
+}
+```
+
+Here's the custom converter for the preceding struct:
+
+```csharp
+public class TemperatureConverter : JsonConverter<Temperature>
+{
+    public override Temperature Read(
+        ref Utf8JsonReader reader,
+        Type typeToConvert,
+        JsonSerializerOptions options)
+    {
+        Debug.Assert(typeToConvert == typeof(Temperature));
+        return Temperature.Parse(reader.GetString());
+    }
+
+    public override void Write(
+        Utf8JsonWriter writer,
+        Temperature value,
+        JsonSerializerOptions options)
+    {
+        writer.WriteStringValue(value.ToString());
+    }
+}
+```
+
+The `[JsonConvert]` attribute on the struct registers the custom converter as the default for properties of type `Temperature`. The converter is automatically used on the `TemperatureC` property of the following type when you serialize or deserialize it:
+
+```csharp
+class WeatherForecastWithTemperatureStruct
+{
+    public DateTimeOffset Date { get; set; }
+    public Temperature TemperatureC { get; set; }
+    public string Summary { get; set; }
+}
+```
+
+## Samples for common scenarios
+
+The following sections provide converter samples that address some common scenarios that built-in functionality doesn't handle.
+
+### Deserialize inferred types to Object properties
+
+When deserializing to a property of type `Object`, a `JsonElement` object is created. The reason is that the deserializer doesn't know what CLR type to create, and it doesn't try to guess. For example, if a JSON element has "true", the deserializer doesn't infer that the value is a `Boolean`, and if an element has "01/01/2019", the deserializer doesn't infer that it's a `DateTime`. You can implement type inference in a custom converter.
+
```

Comment:
The above example about "true" and "datetime" are correct however a more ambiguous example is with a JSON number. If we try to parse as a long (no decimal points) then that may have range issues if the value was originally serialized as a ulong or BigInteger. If we try to parse as a double that will lose precision if the number was originally serialized as a decimal.